teams.md/docs/python/migrations/v1/README.md

```diff
@@ -0,0 +1,507 @@
+
+## Installing Teams AI v2
+
+First, let's install Teams AI v2 into your project. Notably, this won't replace any existing installation of Teams AI v1. When you've completed your migration, you can safely remove the `teams-ai` dependency from your `pyproject.toml` file.
+
+```sh
+uv add microsoft-teams-apps
+```
+
+## Migrate Application class
+
+First, migrate your `Application` class from v1 to the new `App` class.
+
+<table>
+<tr>
+<td> <strong>Teams AI v1</strong> </td> <td> <strong>Teams AI v2</strong> </td>
+</tr>
+<tr>
+<td>
+
+```python
+# in api.py
+from http import HTTPStatus
+
+from aiohttp import web
+from botbuilder.core.integration import aiohttp_error_middleware
+
+from bot import app
+
+routes = web.RouteTableDef()
+
+
+@routes.post("/api/messages")
+async def on_messages(req: web.Request) -> web.Response:
+    res = await app.process(req)
+
+    if res is not None:
+        return res
+
+    return web.Response(status=HTTPStatus.OK)
+
+
+api = web.Application(middlewares=[aiohttp_error_middleware])
+api.add_routes(routes)
+
+# in app.py
+from aiohttp import web
+
+from api import api
+from config import Config
+
+if __name__ == "__main__":
+    web.run_app(api, host="localhost", port=Config.PORT)
+
+# in bot.py
+import sys
+import traceback
+
+from botbuilder.core import TurnContext, MemoryStorage
+from teams import Application, ApplicationOptions, TeamsAdapter
+from teams.state import TurnState
+
+from config import Config
+
+config = Config()
+storage = MemoryStorage()
+app = Application[TurnState](
+    ApplicationOptions(
+        bot_app_id=config.APP_ID,
+        adapter=TeamsAdapter(config),
+        storage=storage
+    )
+)
+
+@app.activity("message")
+async def on_message(context: TurnContext, _state: TurnState):
+    await context.send_activity(f"you said: {context.activity.text}")
+    return True
+
+
+@app.error
+async def on_error(context: TurnContext, error: Exception):
+    # This check writes out errors to console log .vs. app insights.
+    # NOTE: In production environment, you should consider logging this to Azure
+    #       application insights.
+    print(f"\n [on_turn_error] unhandled error: {error}", file=sys.stderr)
+    traceback.print_exc()
+
+    # Send a message to the user
+    await context.send_activity("The bot encountered an error or bug.")
+
+```
+
+</td>
+<td>
+
+```python
+# in main.py
+import asyncio
+import logging
+
+from microsoft.teams.api import MessageActivity
+from microsoft.teams.apps import ActivityContext, App, ErrorEvent
+from microsoft.teams.common import LocalStorage
+
+logger = logging.getLogger(__name__)
+
+# Define the app
+app = App()
+
+# Optionally create local storage
+storage: LocalStorage[str] = LocalStorage()
```

Comment:
doesn't this need to go in the App? also do we need this?

Reply:
nope, I think it's just to demonstrate having local storage outside of the app, was copying TS!
https://microsoft.github.io/teams-ai/typescript/migrations/v1/#migrate-application-class